packages/framework/src/Model/Inquiry/Mail/InquiryMail.php

```diff
@@ -0,0 +1,184 @@
+        protected readonly Setting $setting,
+        protected readonly DomainRouterFactory $domainRouterFactory,
+        protected readonly ImageFacade $imageFacade,
+        protected readonly Domain $domain,
+    ) {
+    }
+
+    /**
+     * @param \Shopsys\FrameworkBundle\Model\Mail\MailTemplate $template
+     * @param \Shopsys\FrameworkBundle\Model\Inquiry\Inquiry $inquiry
+     * @return \Shopsys\FrameworkBundle\Model\Mail\MessageData
+     */
+    public function createMessageForAdmin(MailTemplate $template, Inquiry $inquiry): MessageData
+    {
+        return new MessageData(
+            $this->setting->getForDomain(MailSetting::MAIN_ADMIN_MAIL, $inquiry->getDomainId()),
+            $template->getBccEmail(),
+            $template->getBody(),
+            $template->getSubject(),
+            $this->setting->getForDomain(MailSetting::MAIN_ADMIN_MAIL, $inquiry->getDomainId()),
+            $this->setting->getForDomain(MailSetting::MAIN_ADMIN_MAIL_NAME, $inquiry->getDomainId()),
+            $this->getBodyVariablesReplacementsForAdmin($inquiry),
+            $this->getSubjectVariablesReplacements($inquiry),
+        );
+    }
+
+    /**
+     * @param \Shopsys\FrameworkBundle\Model\Mail\MailTemplate $template
+     * @param \Shopsys\FrameworkBundle\Model\Inquiry\Inquiry $inquiry
+     * @return \Shopsys\FrameworkBundle\Model\Mail\MessageData
+     */
+    public function createMessageForCustomer(MailTemplate $template, Inquiry $inquiry): MessageData
+    {
+        return new MessageData(
+            $inquiry->getEmail(),
+            $template->getBccEmail(),
+            $template->getBody(),
+            $template->getSubject(),
+            $this->setting->getForDomain(MailSetting::MAIN_ADMIN_MAIL, $inquiry->getDomainId()),
+            $this->setting->getForDomain(MailSetting::MAIN_ADMIN_MAIL_NAME, $inquiry->getDomainId()),
+            $this->getBodyVariablesReplacements($inquiry),
+            $this->getSubjectVariablesReplacements($inquiry),
+        );
+    }
+
+    /**
+     * @param \Shopsys\FrameworkBundle\Model\Inquiry\Inquiry $inquiry
+     * @return array<string, string>
+     */
+    protected function getSubjectVariablesReplacements(Inquiry $inquiry): array
+    {
+        return [
+            self::VARIABLE_FULL_NAME => htmlspecialchars($inquiry->getFullName(), ENT_QUOTES),
+            self::VARIABLE_EMAIL => htmlspecialchars($inquiry->getEmail(), ENT_QUOTES),
+            self::VARIABLE_TELEPHONE => htmlspecialchars($inquiry->getTelephone(), ENT_QUOTES),
+            self::VARIABLE_COMPANY_NAME => $this->escapeOptionalString($inquiry->getCompanyName()),
+            self::VARIABLE_COMPANY_NUMBER => $this->escapeOptionalString($inquiry->getCompanyNumber()),
+            self::VARIABLE_COMPANY_TAX_NUMBER => $this->escapeOptionalString($inquiry->getCompanyTaxNumber()),
+            self::VARIABLE_PRODUCT_NAME => htmlspecialchars($inquiry->getProduct()?->getName() ?: '-', ENT_QUOTES),
```

Comment:
`escapeOptionalString` can be used here as well, or?

Reply:
Yes. Yes, it should. 3ac5bdf29d